src/unpack.js

```diff
@@ -0,0 +1,32 @@
+import {isDerivable} from './types';
+import * as util from './util';
+
+/**
+ * dereferences a thing if it is dereferencable, otherwise just returns it.
+ */
+
+export function unpack (thing) {
```

Comment:
I think it is fine to keep `unpack` in a separate file to avoid cyclical dependencies.